gcc/rust/hir/rust-ast-lower.cc

```diff
@@ -735,6 +739,34 @@ ASTLoweringBase::handle_outer_attributes (const HIR::Item &item)
     }
 }
 
+void
+ASTLoweringBase::handle_doc_item_attribute (const HIR::Item &item,
+					    const AST::Attribute &attr)
+{
+  auto simple_doc_comment = attr.has_attr_input ()
+			    && attr.get_attr_input ().get_attr_input_type ()
+				 == AST::AttrInput::AttrInputType::LITERAL;
+  if (simple_doc_comment)
+    return;
+
+  const AST::AttrInput &input = attr.get_attr_input ();
+  bool is_token_tree
+    = input.get_attr_input_type () == AST::AttrInput::AttrInputType::TOKEN_TREE;
+  rust_assert (is_token_tree);
+  const auto &option = static_cast<const AST::DelimTokenTree &> (input);
+  AST::AttrInputMetaItemContainer *meta_item = option.parse_to_meta_item ();
+
+  for (const auto &item : meta_item->get_items ())
+    {
+      std::string value = item->as_string ();
+      if (value.compare ("inline") != 0 && value.compare ("hidden") != 0)
+	{
+	  rust_warning_at (attr.get_locus (), 0,
+			   "unknown %<doc%> attribute %qs", value.c_str ());
+	}
```

Comment:
> I might be misunderstanding something - I know barely nothing about the doc attribute or rustdoc - but aren't many more cases supported? https://doc.rust-lang.org/rustdoc/the-doc-attribute.html
> 
> In that case, we should add a FIXME or handle them all and ignore them since I'm assuming they're mostly handled by rustdoc?

I modelled this behavior after `rustc` itself. When an unknown attribute is encountered in `rustc`, this would happen:

```
warning: unknown `doc` attribute `a`
 --> doc_comment.rs:9:7
  |
9 | #[doc(a,b)] // { dg-warning "unknown .doc. attribute ...." 2 }
  |       ^
  |
  = note: `#[warn(invalid_doc_attributes)]` on by default
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>

warning: unknown `doc` attribute `b`
 --> doc_comment.rs:9:9
  |
9 | #[doc(a,b)] // { dg-warning "unknown .doc. attribute ...." 2 }
  |         ^
  |
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>

warning: 2 warnings emitted
```

Reply:
Hmm. Just checked, `rustc` itself supports more than what's listed in https://doc.rust-lang.org/stable/nightly-rustc/src/rustc_feature/builtin_attrs.rs.html#371.

I guess we will just stub the check for now.

Reply:
The check logic in question is now stubbed.